### PR TITLE
Upgrade to Laravel 5.4.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request upgrades Northstar to use Laravel 5.4!

References [#148973145](https://www.pivotaltracker.com/story/show/148973145).

#### How should this be reviewed?
This follows the steps in the [Upgrade Guide](https://laravel.com/docs/5.4/upgrade#upgrade-5.4.0) (cb51ad4), and then adds a slew of other miscellaneous fixes after tests broke, notably:

🤕 Adds more patches for problems in Laravel-MongoDB, since these changes have not been pulled into the upstream repo. (9a7b5e8, 254e4ac, and more updates in our [patched fork](https://github.com/jenssegers/laravel-mongodb/compare/master...DoSomething:chunkById-5.4)).

📆 Removes some extra hacky stuff we were doing in the model around parsing strange date-times. We no longer store any dates in the database as strings or other weird stuff, so these are unnecessary. Instead, just using a standard mutator on `birthdate` to accept user-formatted dates. (d866488)

📍 Moves routes into `routes/web.php` and `routes/api.php` matching Laravel 5.4+ convention!

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
For now I've switched all the tests to use Taylor's ["BrowserKit" Laravel 5.3 compatibility package](https://github.com/laravel/browser-kit-t…) so we don't have to re-write everything (but we should add a ticket to start updating these, since I'm sure this package won't be maintained forever).

**Comment by weerd** [comment]
image
😱

File: app/Auth/Registrar.php [hunk 1]
```diff
@@ -65,11 +65,11 @@ public function validate(Request $request, User $existingUser = null, array $add
         $existingId = isset($existingUser->id) ? $existingUser->id : 'null';
         $rules = [
             'first_name' => 'max:50',
-            'email' => 'email|unique:users,email,'.$existingId.',_id|required_without:mobile',
-            'mobile' => 'mobile|unique:users,mobile,'.$existingId.',_id|required_without:email',
-            'birthdate' => 'date',
-            'country' => 'country',
-            'password' => 'min:6|max:512',
+            'email' => 'email|nullable|unique:users,email,'.$existingId.',_id|required_without:mobile',
```
Comment (mendelB) [inline app/Auth/Registrar.php]: Does this mean we can now re-add the ConvertEmptyStringsToNull middleware in aurora?
https://github.com/DoSomething/aurora/pull/173/files

Comment (DFurnes) [reply app/Auth/Registrar.php]: Yes!

**Pull request merged**